openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -205,3 +212,80 @@ def award_program_certificates(self, username):
         LOGGER.info('User %s is not eligible for any new program certificates', username)
 
     LOGGER.info('Successfully completed the task award_program_certificates for username %s', username)
+
+
+def post_course_certificate(client, username, certificate):
+    """
+    POST a certificate that has been updated to Credentials
+    """
+    client.credentials.post({
+        'username': username,
+        'status': 'awarded' if certificate.is_valid() else 'revoked',  # Only need the two options at this time
+        'credential': {
+            'course_run_key': str(certificate.course_id),
+            'mode': certificate.mode,
+            'type': COURSE_CERTIFICATE,
+        }
+    })
+
+
+@task(bind=True, ignore_result=True, routing_key=ROUTING_KEY)
+def award_course_certificates(self, username, course_run_key):
+    LOGGER.info('Running task award_course_certificates for username %s', username)
+
+    countdown = 2 ** self.request.retries
+
+    # If the credentials config model is disabled for this
+    # feature, it may indicate a condition where processing of such tasks
+    # has been temporarily disabled.  Since this is a recoverable situation,
+    # mark this task for retry instead of failing it altogether.
+
+    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
+        LOGGER.warning(
+            'Task award_course_certificates cannot be executed when credentials issuance is disabled in API config',
+        )
+        raise self.retry(countdown=countdown, max_retries=MAX_RETRIES)
```

Comment:
is_learner_issuance_enabled was already checked in the handler. Might not need to check a second time? I guess it doesn't hurt to be careful.

Even if we do check here, I doubt we need to bother retrying, though.

Reply:
Cool I'll drop both. I think the retry occurred in both places in the other job because situations where some one was toggling things off/on while jobs were running might make sense to try again in a bit... but agreed probably not a super real scenario and we're already checking once.

Reply:
Ehhh... gonna keep it I decided, the other job has it as well as the comment which I (shamelessly) stole as well,

```
    # If the credentials config model is disabled for this
    # feature, it may indicate a condition where processing of such tasks
    # has been temporarily disabled.  Since this is a recoverable situation,
    # mark this task for retry instead of failing it altogether.
```

Reply:
Ah yeah. The comment right above. I should have read that. :)